### ISSUE TITLE
Spawning/Joining an ongoing round sets the character's wallet money to zero.

### ISSUE THREAD

**Issue by Sarynkia** [FakeFishGames/Barotrauma#13086]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Disclaimer for the disclaimer: we were running texture replacement mods (which I don't _believe_ is related to the issue) but that's all. I checked the "My issue happened while using mods" because of this.

I believe that if a character joins mid-session, with the server running a save file in which said character did not die, their wallet money should be retained. (Is this intended behaviour?)
However, once the character decides to spawn in without waiting for a round to end first, it does not happen.

Said character appears to retain skills.

### Reproduction steps

1. Have a multiplayer save with a character having non-zero wallet money.
2. Have the client with said character join mid-round.
3. Profit. Client with said character would see their wallet money set to zero. However, they do not appear to lose skill points as with actual character death-respawns

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.1.19.3 (Treacherous Tides Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! Fixed in https://github.com/Regalis11/Barotrauma-development/commit/9f2fac58f797d0609e732f07a167da2c67b7d557 (commit in our private repo)

**Comment by Jasontti** [comment]
Tested in: https://github.com/Regalis11/Barotrauma-development/pull/4703/commits/7983d27ab03d90acb04000360ba3a691068b9d30

Joining mid round does retain previous amount of money.

Issue found, if character with money dies during round and somebody touches the corpse, money will be transferred to bank or wallet, but if the dead player chooses to respawn they will still have same amount of money. Effectively a money dupe.

**Comment by Regalis11** [comment]
Nice catch, wouldn't have occurred to me to test that! Addressed in https://github.com/Regalis11/Barotrauma-development/pull/4703/commits/d41c1374774fea58cdd4ed9b28679b571a6932b0

**Comment by Jasontti** [comment]
Tested in: https://github.com/FakeFishGames/Barotrauma-development/pull/4703/commits/d41c1374774fea58cdd4ed9b28679b571a6932b0

Dublication could not be replicated any more.

How ever, Looted money from dead players is not saved when round ends

**Comment by Regalis11** [comment]
I couldn't repro the issue. Tested by doing the following:
1. Two clients on a server
2. Move 250 mk to client A, 3250 mk left in bank
3. Disconnect client A, kill their character and loot the money as client B (there's now 3500 mk in the bank)
4. Start a new round, there's still 3250 mk in the bank.

**Comment by Jasontti** [comment]
Steps to repro:
1. 2 Clients on listen server
2. Give some money to client in outpost
3. Start a round
4. Kill the client and choose "i'll wait" option
5. Loot the body as host
6. End the round

**Comment by Jasontti** [comment]
Tested in: https://github.com/FakeFishGames/Barotrauma-development/pull/4703/commits/46674bb73b0ef13084fba163b7206faef1d851b3

Tested the above case and that works, but there are other issues.

Looting dead players body gives money and respawning in shuttle player still has money.
Looks like issue from 2 iterations ago has returned.
1. 2 players, 1 has money
2. money player dies
3. other loots and receives the money
4. after respawn dead player also has the same money

After going to a new round with 2 clients and one has some money, i was able to repeatedly grab the killed person to gain amount of held money each time. Default destination does not matter.
1. 2 players, 1 has money
2. money player dies
3. other loots and receives the money
4. other loots again to receive more money
5. ???
6. profit

**Comment by Regalis11** [comment]
Couldn't find a way to repro. :/ Could it be that you didn't have `multiclienttestmode` enabled? If the server got confused about which character (and wallet) belongs to which character, it could explain the issue you encountered. Here's a clip of how I tested this:
https://github.com/FakeFishGames/Barotrauma/assets/14351070/f8c771b4-8f12-40d1-9a9c-d3f8ee4d456f

**Comment by Jasontti** [comment]
Tested again with multiple steam accounts and could repro and have found the additional info.

Dube happens when character has money from previous round on them.
I moved 500mk to the client from bank in outpost and then started the round.
If you give money in new round so that they have 1000mk for example.
you get that 1000mk from first grab and 500mk from next ones.

https://github.com/FakeFishGames/Barotrauma/assets/8488224/61315546-e829-4a87-b3b9-b926372edebe
